### PR TITLE
{Test} Fix some live tests for acr, appservice, backup, netappfiles, batch

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

1. Add sleep between `az acr identity assign` and `az acr identity remove` as operations on identity are async.
2. backup vault are now soft-delete enabled by default and can not be deleted when there are soft-deleted items in it. Adding an option in the `VaultPreparer` to disable soft delete and allow the vault delete to fail in soft-delete-enabled cases.
3. netappfiles has a quota of 10 accounts per subscription, make some tests run in serial to reduce the number of accounts created in parallel.
4. use ResourceGroupPreparer in batch test and fix outdated key vault related test.
 
**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Test

File: src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_data_plane_commands.py [hunk 1]
```diff
@@ -52,11 +48,12 @@ def test_batch_certificate_cmd(self, resource_group, batch_account_name):
             self.check('thumbprintAlgorithm', 'sha1'),
             self.check('state', 'deleting')])
 
-    # Since this test requires core quota, we must statically create an account with quota
+    @ResourceGroupPreparer()
+    @BatchAccountPreparer()
     def test_batch_pool_cmd(
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/batch/tests/latest/test_batch_data_plane_commands.py:53]: Change to use resource preparers and you can use environment variables `AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME` and `AZURE_CLI_TEST_DEV_BATCH_ACCT_NAME` for existing resources in live tests when necessary.

**Comment by fengzhou-msft** [comment]
@bgklein can you help review the changes for batch tests? This is to fix the failures in our weekly [live tests](https://dev.azure.com/azure-sdk/internal/_build?definitionId=1896&_a=summary) validation. The result from last week for batch is here: https://clitestresultstac.blob.core.windows.net/20200918224751live/batch.report.parallel.html

**Review by Juliehzl — APPROVED** [review]
Approve for my code owner part

**Review by yungezz — APPROVED** [review]

**Pull request merged**